internal/proxy/reverseproxy/reverseproxy.go

```diff
@@ -50,4 +68,50 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		logger.Warnf("Error %q - code: %d", rr.err, rr.statusCode)
 		http.Redirect(rw, req, rp.RedirectURL.String(), http.StatusFound)
 	}
+
+	if rr.statusCode == http.StatusForbidden {
```

Comment:
If not logged in jenkins sets status code as `Forbidden`, have not seen other 4xx status for far.

Reply:
probably a good idea to check for any 4xx status

Reply:
like 
```suggestion
	if rr.statusCode >= 400 && rr.statusCode < 500 {
```